chefkoch/recipe.py

```diff
@@ -0,0 +1,771 @@
+        """
+        content = "The flavour file contains: "
+        for key in self:
+            content += "\n  " + self[key].tostring()
+        return content
+
+    # todo overwrite return of the parameter values list so that
+    # flavour["param_name"] returns flavour["param_name"][0] if there is only
+    # one element in the list
+
+
+class FileParamValue:
+    """
+    A single parameter value which is a file, defined by filepath and optional
+    key/passphrase to the file.
+    """
+
+    key = ""   #: Saves optional passphrase to the given file, defaults to "".
+    file = ""  #: Saves the filepath, if the given file exists, defaults to "".
+
+    def __init__(self, filepath, key):
+        """
+        Initialises a new parameter value, that holds a file.
+
+        :param filepath: file path as given in flavour.json
+        :type filepath: string
+        :param key: optional key or passphrase to the file
+        :type key: string or None
+        :raises IOError: In case, the filepath is None or the path does not
+            exist, there is an IOError raised and the functions returns.
+        """
+        logger.debug("Creating new FileParamValue")
+        logger.debug("Filepath (dict): " + str(filepath))
+        logger.debug("Key (dict): " + str(key))
+        self.key = key
+        logger.debug("Key: " + str(self.key))
+        if filepath is None:
+            raise IOError("The filepath is None.")
+            return
+        if os.path.isfile(filepath):
+            self.file = filepath
+            logger.debug("Filepath: " + str(self.file))
+        else:
+            logger.warning("The file " + filepath + " does not exist.")
+            raise IOError("The file " + filepath + " does not exist.")
+            return
+
+    def tostring(self):
+        """
+        Returns a printable and formatted string that shows the
+        FileParamValue and its values.
+
+        :returns:
+            string - Holds structured information on the FileParamValue
+        """
+        content = "Value is following file: \n  "
+        content+= self.file + "\n  Key: " + str(self.key)
+        return content
+
+
+class Param:
+    """
+    A parameter with all values attached to it.
+    """
+    # todo: make Param class extend list so that Param returns Param.values
+    values = [] #: All possible values of the parameter
+    
+    def __init__(self, name, entry):
+        """
+        Creates a new paramter from the JSON data gotten from the flavour file.
+        
+        :param name:        name as provided as in flavour file
+        :type name:         string
+        :param entry:       if the falvour file was a dict, it was flavour['name']
+        :type entry:        dictionary
+        """
+        self.values = []
+        logger.debug("Creating a new parameter " + str(name))
+        self.name = name
+        if type(entry) is not list: # a single value so to say
```

Comment:
is that sufficient?
Since you also could extend the list type, it is beneficiary to test for inheritance from lists (and also do not forget tuples and sets) using `if isinstance(entry, (list, tuple, set)):`